src/mono/mono/metadata/class.c

```diff
@@ -1379,6 +1379,129 @@ mono_method_set_generic_container (MonoMethod *method, MonoGenericContainer* con
 	mono_image_property_insert (mono_method_get_image (method), method, MONO_METHOD_PROP_GENERIC_CONTAINER, container);
 }
 
+/**
+ * mono_method_set_verification_success:
+ *
+ * Sets a bit indicating that the method has been verified.
+ *
+ * LOCKING: acquires the image lock.
+ */
+void
+mono_method_set_verification_success (MonoMethod *method)
+{
+	g_assert (!method->is_inflated);
+
+	mono_image_property_insert (mono_method_get_image (method), method, MONO_METHOD_PROP_VERIFICATION_SUCCESS, GUINT_TO_POINTER(1));
+}
+
+/**
+ * mono_method_get_verification_sucess:
+ *
+ * Returns \c TRUE if the method has been verified successfully.
+ *
+ * LOCKING: acquires the image lock.
+ */
+gboolean
+mono_method_get_verification_success (MonoMethod *method)
+{
+	if (method->is_inflated)
+		method = ((MonoMethodInflated *)method)->declaring;
+
+	gpointer value = mono_image_property_lookup (mono_method_get_image (method), method, MONO_METHOD_PROP_VERIFICATION_SUCCESS);
+
+	return value != NULL;
+}
+
+/**
+ * mono_method_lookup_infrequent_bits:
+ *
+ * Looks for existing \c MonoMethodDefInfrequentBits struct associated with
+ * this method definition.
```

Comment:
I'd consider specifically putting that this will *not* allocate a new struct just to make the difference clear.